docs/dev/go-style-guide.md

```diff
@@ -117,3 +118,200 @@ sdkerrors.Wrapf(
   <value 2>
 )
 ```
+
+## Common Mistakes
+
+This is a compilation of some of the common mistakes we see in the repo that should be avoided.
+
+---
+Keep receiver names short [Details here](https://google.github.io/styleguide/go/decisions#receiver-names)
+
+```go
+// bad
+func (chain *TestChain) NextBlock() {
+    res, err := chain.App.FinalizeBlock(&abci.RequestFinalizeBlock{
+        Height:             chain.ProposedHeader.Height,
+        Time:               chain.ProposedHeader.GetTime(),
+        NextValidatorsHash: chain.NextVals.Hash(),
+    })
+    require.NoError(chain.TB, err)
+    chain.commitBlock(res)
+}
+```
+
+```go
+// good
+func (c *TestChain) NextBlock() {
+    // Ommitted
+```
+
+---
+**Naked returns**
+
+Only acceptable in a small function. Once it's a medium-sized function, be explicit with your returned values. Similarly, do not name result parameters just because it enables you to use naked returns. Clarity is always more important than saving a few lines in your function.
+
+---
+**Struct defined outside of the package**
```

Comment:
I def agree with this, but is this a common issue you've seen (it very well might be, I just can't remember seeing it)?